### DISCUSSION TITLE
2 heaters configurated but only one is ON, no alternate sequences as described into documentation

### DISCUSSION THREAD

**Discussion by jolypas** [jmcollin78/versatile_thermostat#372] [Q&A]
I have a quite big room with 2 heaters at each end.
Most of the time only heater 1 is active which is problem even if the temperature sensor is in the middle : one side is cooler (and only one heater most of the time is sollicated)

Anyway to modify this behaviour : like define the time to swicth between heaters when needed 👍 
![image](https://github.com/jmcollin78/versatile_thermostat/assets/14258266/a4874899-01ed-4f4c-987d-4e2db2d5d569)

**Comment by MBRetrofit** [comment]
You can try Central Configuration settings like the default.

**Reply by jolypas** [reply]
thanks for recommendation !
Seems better, will monitor on longer period

**Comment by jmcollin78** [comment]
@jolypas ,

Please post your Vtherm attributes (Dev. Tools / States). This should look like this: https://github.com/jmcollin78/versatile_thermostat/blob/main/README-fr.md#attributs-personnalis%C3%A9s

I seems not normal even with or without central configuration.

**Comment by jolypas** [comment]
Hello

> hvac_modes: heat, off
min_temp: 15
max_temp: 25
target_temp_step: 0.1
preset_modes: none, frost, eco, comfort, boost, activity
current_temperature: 19.9
temperature: 20
hvac_action: heating
preset_mode: boost
is_on: true
hvac_mode: heat
type: null
is_controlled_by_central_mode: true
last_central_mode: null
frost_temp: 15
eco_temp: 17
boost_temp: 20
comfort_temp: 19
frost_away_temp: 15
eco_away_temp: 17
boost_away_temp: 21
comfort_away_temp: 20
power_temp: 13
target_temperature_step: 0.1
ext_current_temperature: 10.64
ac_mode: false
current_power: null
current_power_max: null
saved_preset_mode: boost
saved_target_temp: 20
saved_hvac_mode: null
motion_sensor_entity_id: binary_sensor.0x00158d0001232782_occupancy
motion_state: off
power_sensor_entity_id: null
max_power_sensor_entity_id: null
overpowering_state: null
presence_sensor_entity_id: binary_sensor.0xa4c1388e99cd04b6_presence
presence_state: on
window_state: off
window_auto_state: off
window_bypass_state: false
window_sensor_entity_id: binary_sensor.ouvertures_salon
window_delay_sec: 30
window_auto_enabled: false
window_auto_open_threshold: 3
window_auto_close_threshold: 0
window_auto_max_duration: 30
window_action: window_turn_off
security_delay_min: 60
security_min_on_percent: 0.5
security_default_on_percent: 0.1
last_temperature_datetime: 2024-02-03T17:31:05.857966+01:00
last_ext_temperature_datetime: 2024-02-03T17:26:53.778056+01:00
security_state: false
minimal_activation_delay_sec: 10
device_power: 1
mean_cycle_power: 0.15
total_energy: 162.61
last_update_datetime: 2024-02-03T17:31:33.842045+01:00
timezone: Europe/Paris
temperature_unit: °C
is_device_active: true
ema_temp: 19.92
is_used_by_central_boiler: false
is_over_switch: true
is_inversed: false
underlying_switch_0: switch.radiateur_1
underlying_switch_1: switch.radiateur_2
underlying_switch_2: null
underlying_switch_3: null
on_percent: 0.15
power_percent: 15
on_time_sec: 46
off_time_sec: 253
cycle_min: 5
function: tpi
tpi_coef_int: 0.6
tpi_coef_ext: 0.01
friendly_name: Thermostat Salon
supported_features: 17

**Reply by jmcollin78** [reply]
ok. This seems good. I don't have any explanation on the behavior in the first screen. You should have something very regular like this: ![Capture d’écran 2024-02-03 à 18 06 24](https://github.com/jmcollin78/versatile_thermostat/assets/1712374/8db9757d-5e4d-4c2c-914b-3847b6c11466) and really not this: ![Capture d’écran 2024-02-03 à 18 06 56](https://github.com/jmcollin78/versatile_thermostat/assets/1712374/d0768ddc-bd60-485a-91b5-8fe3de39de55)

Each 5 min (your cycle duration) it should change.

This is like VTherm is not controling your heaters at all. VTherm cannot do that for me.

**Reply by jolypas** [reply]
> This is like VTherm is not controling your heaters at all.

I don't really understand  what could control these heaters as I have only VTherm to pilot them and the heater 1 is reacting to VTherm commands
In the other rooms with only one heater I have no issue.

All these heaters are connnected using Qubino "fil pilote"  through zwaveJS, could be the root cause ?

**Comment by MBRetrofit** [comment]
Try Developer Tools/YAML/YAML Configuration Reloading/Versatile Thermosta Configuration - Reset

**Reply by jolypas** [reply]
Hi, not sure to understand how to "reset" the configuration using "developper tool/yaml". I can just click to "reload" configuration :
![image](https://github.com/jmcollin78/versatile_thermostat/assets/14258266/07d32980-515a-4cdd-841d-115b7781c7e4)

**Reply by jmcollin78** [reply]
Yes you can try this or restart HA.
Maybe something is not well loaded. Have you look at the logs ?

**Comment by jolypas** [comment]
Finally, I created a group with the 2 heaters, not perfect but better that only one heater active

**Reply by jmcollin78** [reply]
yes, but we should rather find what is wrong. I guess this is in your environment or configuration because it is working either.

Can you install the last 5.4.1 release, activate the log in INFO mode (cf README if needed), configure the VTherm with both heater (and not the group) and send me the history of heater activations and the logs.

We have to find what is wrong. (May be it is a hidden bug...)

**Reply by jolypas** [reply]
OK, what an "after sales" service ! Really appreciated.
I just upgrade to 5.4.1 (HACS was not proposing any updates so i was in 5.3.3) 
I cannot find the word "log" into the readme. I know how to configure some log differently inot configuation.yaml but what key word to use for Vtherm ?

logger:
  default: error
  logs:
    rflink: warning
    homeassistant.components.rflink.protocol: debug

**Reply by jmcollin78** [reply]
I was sure I have done it but no. This is now done: https://github.com/jmcollin78/versatile_thermostat/blob/main/README-fr.md#activer-les-logs-du-versatile-thermostat

**Comment by jolypas** [comment]
one remark into HACS, the available release is 5.3.3 , strange :
![image](https://github.com/jmcollin78/versatile_thermostat/assets/14258266/75caab63-7b2f-4421-b656-1763d485e455)

**Reply by jmcollin78** [reply]
HACS update its information once per day I think. Because the release was just published, I suppose it is not updated yet.

**Comment by jolypas** [comment]
attached the logs ( 2 heaters for Thermostat Salon)
[versatile.log](https://github.com/jmcollin78/versatile_thermostat/files/14156264/versatile.log)

 and the graph 👍 
![image](https://github.com/jmcollin78/versatile_thermostat/assets/14258266/86a78a7b-cb4b-4d70-ba4e-22152f6b1692)

**Comment by jmcollin78** [comment]
Ok, it is clear. Your presence sensor is broken. It always change state every seconds, which restart the loop (and then you have only the first heater which is heating):
```

2024-02-04 10:44:20.846 INFO (MainThread) [custom_components.versatile_thermostat.base_thermostat] VersatileThermostat-Thermostat Salon - Presence changed. Event.new_state is <state binary_sensor.0xa4c1388e99cd04b6_presence=on; detection_delay=0.1, device=applicationVersion=74, dateCode=, friendlyName=Présence Salon, hardwareVersion=1, ieeeAddr=0xa4c1388e99cd04b6, manufacturerID=4417, manufacturerName=_TZE204_qasjif9e, model=ZY-M100-S_2, networkAddress=46681, powerSource=Mains (single phase), stackVersion=0, type=Router, zclVersion=3, fading_time=60, illuminance_lux=102, linkquality=255, maximum_range=9.5, minimum_range=0.6, presence=True, radar_sensitivity=7, target_distance=4.41, device_class=presence, friendly_name=Présence Salon Présence @ 2024-02-04T10:33:26.859285+01:00>, _attr_preset_mode=none, activity=activity

```

I've got this log every second. I think you have misunderstood presence and movement.
I guess the sensor `binary_sensor.0xa4c1388e99cd04b6_presence` is a movement sensor.

What you should configure in VTherm is an occupancy sensor 'is there someone at home' and not 'is there someone moving'.

Maybe you have to inverse the both sensor. 
presence_sensor_entity_id: binary_sensor.0xa4c1388e99cd04b6_presence
motion_sensor_entity_id: binary_sensor.0x00158d0001232782_occupancy

**Reply by jmcollin78** [reply]
Your movement sensor seems configured with 0.1 sec of detection which make it trigger all the time when someone is moving. 30 sec is certainly a better value.

**Reply by jolypas** [reply]
yes I saw the issue into the logs

I removed movement & presence sensors for the moment and it is OK for a first step.
In fact presence is an aqua "radar" and movment a PIR 
I need to further investigate, I know how to debug now thanks to your readme update  ;-) 

Thanks for your time, you win a beer !

This versatile thermostat is a must , as heating is the most critical into my domotic and I don't have the right to mistake ( WAF !)

**Reply by jmcollin78** [reply]
> Thanks for your time, you win a beer !

Ooooh thank you for this ! It is very pleasing 👍
